### PR TITLE
Add user collection

### PR DESCRIPTION
**Pull request by sakirdemirhan**
Add user collection after register or login with Google.
Syncs Firebase user and user table after login.

### CODE REVIEW

**Review by writeriks — COMMENTED** [review]
Can you create a hook that would return the user

**Review by writeriks — COMMENTED** [review]
isAuthenticated selector is not necessary anymore.

**Review by writeriks — COMMENTED** [review]
We need check for email and username for every registered users, so that we know if name is taken or not

**Review by writeriks — COMMENTED** [review]
Please add full name field on register screen

File: src/services/firebase-auth-service/firebase-auth-service.ts [hunk 1]
```diff
@@ -21,6 +26,25 @@ class FirebaseAuthService {
...
+        // When sign in with Google, email is automatically verified. Need to update in user collection
+        await userService.syncUser(user, userDocument);
+      } else {
+        // If new user, add user to the user collection
+        const customUser: User = {
+          [UserEnum.USER_ID]: user.uid,
+          [UserEnum.CREATED_AT]: Timestamp.now(),
+          [UserEnum.EMAIL]: user.email as string,
+          [UserEnum.USERNAME]: user.email as string,
+          [UserEnum.IS_EMAIL_VERIFIED]: user.emailVerified,
```
Comment (aytacgedik) [inline src/services/firebase-auth-service/firebase-auth-service.ts:42]: verification'i boyle yapmak riskli. farkli bi sekilde verify edelim, userlarin js degistirip verify olmus acmalarina izin vermeyek direk.

Comment (writeriks) [reply src/services/firebase-auth-service/firebase-auth-service.ts:43]: Lets do this in another PR

**Review by writeriks — APPROVED** [review]

**Pull request merged**